### DISCUSSION TITLE
Carousel: Multiple Items inside carousel

### DISCUSSION THREAD

**Discussion by clojurelife** [AvaloniaUI/Avalonia#9048] [Q&A] [answered]
I'm new to Avalonia. I have a working Carousel, but I need to bind multiple Items inside each Carousel Transition. I understand that a Carousel is just an ItemsControl and I thought I could just set ItemsCount to have multiple items like this: `                <Carousel Name="carousel" Margin="0,40,0,0" Items="{Binding Items}"  ItemsCount="4">`, but that doesn't work. In fact I get this error: `Unhandled exception. System.MethodAccessException: Attempt by method 'dashBoardCarousel.Views.DashboardWindow.!XamlIlPopulate(System.IServiceProvider, dashBoardCarousel.Views.DashboardWindow)' to access method 'Avalonia.Controls.ItemsControl.set_ItemCount(Int32)' failed.
` Thanks ahead of time.

**Comment by timunie** [comment, marked as answer]
I don't get what you are trying to do. You can have as many items you like. But items count is automatically determined by items.Count

**Reply by clojurelife** [reply]
So right now I only have one item in my Carousel, this is my Dashboard.axaml: 
`        <Grid Grid.Row="1" 
              ColumnDefinitions="Auto,*,Auto" MaxWidth="1800" HorizontalAlignment="Stretch" Margin="0 1 0 0">
            <Button ZIndex="1" Name="left" Grid.Column="0" VerticalAlignment="Center" Padding="10,20" Margin="4">
                <Path Data="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z" Fill="Black"/>
            </Button>
            <Panel Grid.Column="1" DataContext="{Binding List}" HorizontalAlignment="Center" VerticalAlignment="Center">
                <Carousel Name="carousel" Margin="0,20,0,0" Items="{Binding Items.Count}" >
                    <Carousel.PageTransition>
                        <PageSlide Duration="0.25" Orientation="Horizontal" />
                    </Carousel.PageTransition>
                    <ItemsControl.ItemsPanel>
                    <ItemsPanelTemplate>
                        <WrapPanel />
                    </ItemsPanelTemplate>
                </ItemsControl.ItemsPanel>
                    <ItemsControl.ItemTemplate>
                    <DataTemplate>
                        <views:AppView Margin="0,0,20,20"/>
                    </DataTemplate>
                </ItemsControl.ItemTemplate>
                </Carousel>
            </Panel>
            <Button Name="right" Grid.Column="2" VerticalAlignment="Center" Padding="10,20" Margin="4">
                <Path Data="M4,11V13H16L10.5,18.5L11.92,19.92L19.84,12L11.92,4.08L10.5,5.5L16,11H4Z" Fill="Black"/>
            </Button>
        </Grid>`
And this is my AppView.axaml:
`    <StackPanel Spacing="5" Width="150">
        <Border CornerRadius="10" ClipToBounds="True">
            <Panel Background="{DynamicResource HeaderFooterBlueBrush}">
                <TextBlock Height="150" Text="{Binding Description}" HorizontalAlignment="Center" />
            </Panel>
        </Border>
    </StackPanel>`
I would like to up to 8 items in each Carousel, because the items are dynamic. I'm not sure where i would set Items.Count at? Thanks for responding.

**Reply by timunie** [reply]
Ah so you have, I don't know, let's say 20 items and don't want all of them in one Carousel? 

Do you want this layout?:
- First Carousel (Items 0-7)
- Second Carousel (Items 8-15)
- Third Carousel (Items 16-19)
- [Add more Carousels if needed]

In this case you need to find a way to paginate the data. This is a task for your `ViewModel` and not for the UI. Maybe this helps you further: https://www.codeproject.com/Articles/5284771/Data-Paging-with-Dynamic-Data

**Reply by clojurelife** [reply]
thank you so much for the help @timunie, i kind of thought that was the case, but it seems i can produce almost the look i need with a wrap and go on to work on more important stuff. Thanks again.
